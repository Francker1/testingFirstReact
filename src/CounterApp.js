
import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 10 }) => {

    const [ counter, setCounter ] = useState(value);

    const handleAdd = () => setCounter( counter + 1 );

    const handleSubs = () => setCounter( counter -1 );

    const handleReset = () => setCounter(value);
    

    return (
      <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>

        <button onClick={ handleAdd } className="btn btn-plus">+1</button>
        <button onClick={ handleReset } className="btn btn-reset">Reset</button>
        <button onClick={ handleSubs }  className="btn btn-sub">-1</button>
      </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number,
}

export default CounterApp;
