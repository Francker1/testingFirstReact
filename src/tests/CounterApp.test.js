import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe("Pruebas en contador", () => {
    

    let wrapper = shallow(<CounterApp />); //undefined -> asi mantengo el Intellisense del elemento

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test("debe mostrar CounterApp correctamente", () => {

        const wrapper = shallow(<CounterApp value={256} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("debe mostrar numero por defecto: 10", () => {

        const numDefault = wrapper.find("h2").text();

        expect(numDefault).toBe("10");
    });

    test("debe incrementar contador al click +1", () => {

        wrapper.find("button").at(0).simulate("click");
        const btn1Text = wrapper.find("h2").text().trim();

        expect(btn1Text).toBe("11");
    });

    test("debe decrementar contador al click -1", () => {

      wrapper.find("button").at(2).simulate("click");
      const btn1Text = wrapper.find("h2").text().trim();

      expect(btn1Text).toBe("9");
    });

    test("debe resetear a valor dado 256", () => {

        const wrapper = shallow(<CounterApp value={105} />);

        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(1).simulate("click");
        const btn1Text = wrapper.find("h2").text().trim();

        expect(btn1Text).toBe("105");
    });
})

