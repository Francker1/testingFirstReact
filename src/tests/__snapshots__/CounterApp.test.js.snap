// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en contador debe mostrar CounterApp correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    256
  </h2>
  <button
    className="btn btn-plus"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="btn btn-reset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn btn-sub"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
